casper/src/main/scala/coop/rchain/casper/engine/Initializing.scala

```diff
@@ -98,6 +97,9 @@ class Initializing[F[_]
               s"Valid approved block ${PrettyPrinter.buildString(block, short = true)} received. Restoring approved state."
             )
 
+        // Record approved block in DAG
+        _ <- BlockDagStorage[F].insert(block, invalid = false, approved = true)
```

Comment:
This insert should be after line 108 where is insert to BlockStore. We don't want to save approved block before LFS state is completely downloaded.

Reply:
It's not Approved block its just normal block insertion, same as done inside `requestApprovedState`, but insert with approved flag. `LastApprovedBlock[F].set(approvedBlock)` is on line 109, without it restoring will be not complete and start again after restart.